web-console/4.0-designs/binding/binding.md

```diff
@@ -0,0 +1,30 @@
+# Binding
+
+## Create binding
+![](img/binding-prompt.png)
+- New service instances will have a section at the top that describe what service bindings are for and how to use them. The box will also contain a button to create a service binding for the service instance.
+- This box should disappear once it has been dismissed or a binding has been created.
+![](img/create-binding.png)
+- Clicking on the create button enables the user to select a name for their binding and approve its creation.
+
+## Add secret to application
+![](img/binding-pending.png)
+- Once the binding has been created, the user will be taken to the service bindings tab of the service instance.
+- When only a single binding has been created, this tab will contain a similar box with information about service bindings.
+- While the binding secret has not yet been created, the box should contain no button, but instead a notice that the binding is not ready.
+
+![](img/binding-ready.png)
+- Once the binding is ready and the secret has been created, the pending notice should be replaced with a button that enables users to add the secret to an application. At this point, the same action will become enabled in the kebab menu of the binding as well.
+- While the box is present, the list toolbar should be obscured, but it should become available when the box is dismissed, either through the close icon or by activating the `Add Secret to Application` action.
+
+![](img/add-secret.png)
+- Taking the action to add a secret to an application will produce a modal that enables the user to select which application to bind to and to provide an optional prefix for the secret.
```

Comment:
I think we still want to have volume as an option. If the secret contains sensitive data like passwords, mounting it as a volume is more secure than using environment variables. There are other cases where you might want it as a file as well.

Reply:
Will update